### PR TITLE
Delyno hire even harder

### PR DESCRIPTION
**Pull request by demainou**
Added the Data Visualizations Engineer task for review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Requirements and notes:
 You can use any tool to produce the output, e.g. Python, R, Excel, Power BI, Tableau, etc. The final deliverable needs to be a pdf report with your analysis.
 
 ## How to submit
-### Candidates where programming is required (Data Scientist and Engineers)
+### Candidates where programming is required (Data Scientist;  Engineers and Visualization Engineers)
```
Comment (Gordonei) [inline README.md]: For the sake of consistency - visualization -> visualisation

File: README.md [hunk 2]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+### 6. Data Visualization Task (if applying for a Data Visualization Engineering Position)
+
+Using the `sr_hex_truncated.csv` dataset and open source web development technologies, develop a data visualization / dashboard that help to answer the question:
+
+*"In which suburbs should the Water and Sanitation directorate concentrate their infrastructure improvement efforts?".*
+
+The data visualization / dashboard must include the following:
+
+1.    A chart (plot) or charts (plots) that helps to answer the above question.
+2.    A spatial map that helps to answer the above question.
```
Comment (Gordonei) [inline README.md]: A spatial map -> A cartographic map with identifiable landmark features (e.g. major roads, railways, etc.)

File: README.md [hunk 3]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
    * temporal accuracy to within 6 hours
 Please also remove any columns which you believe could lead to the resident who made the request being identified. We expect in the accompanying report that you will justify as to why this data is now anonymised. Please limit this commentary to less than 500 words. If your code is written in a code notebook such as Jupyter notebook or Rmarkdown, you can include this commentary in your notebook.
 
+### 6. Data Visualization Task (if applying for a Data Visualization Engineering Position)
+
+Using the `sr_hex_truncated.csv` dataset and open source web development technologies, develop a data visualization / dashboard that help to answer the question:
+
+*"In which suburbs should the Water and Sanitation directorate concentrate their infrastructure improvement efforts?".*
+
+The data visualization / dashboard must include the following:
```
Comment (Gordonei) [inline README.md]: viz -> vis

File: README.md [hunk 4]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
    * temporal accuracy to within 6 hours
 Please also remove any columns which you believe could lead to the resident who made the request being identified. We expect in the accompanying report that you will justify as to why this data is now anonymised. Please limit this commentary to less than 500 words. If your code is written in a code notebook such as Jupyter notebook or Rmarkdown, you can include this commentary in your notebook.
 
+### 6. Data Visualization Task (if applying for a Data Visualization Engineering Position)
+
+Using the `sr_hex_truncated.csv` dataset and open source web development technologies, develop a data visualization / dashboard that help to answer the question:
```
Comment (Gordonei) [inline README.md]: open source web development technologies -> open source web technologies (this is a bit of a nitpick)

viz -> vis

File: README.md [hunk 5]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+
+*"In which suburbs should the Water and Sanitation directorate concentrate their infrastructure improvement efforts?".*
+
+The data visualization / dashboard must include the following:
+
+1.    A chart (plot) or charts (plots) that helps to answer the above question.
+2.    A spatial map that helps to answer the above question.
+3.    Make (1) and (2) interactive to allow users to explore the data and uncover insights
+4.    Cross filtering: a filter on the map must update the chart(s) accordingly with the filter, and vice versa.
+5.    Data Storytelling: use the dataset to create a data-driven story, using visualizations to support your narrative.
```
Comment (Gordonei) [inline README.md]:
```suggestion
5.    Data Storytelling: in a separate markdown document, use the dataset and your visualisations to outline a data-driven story that answers the above question. In your document, describe how your visualisations support your narrative.
```

File: README.md [hunk 6]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+Using the `sr_hex_truncated.csv` dataset and open source web development technologies, develop a data visualization / dashboard that help to answer the question:
+
+*"In which suburbs should the Water and Sanitation directorate concentrate their infrastructure improvement efforts?".*
+
+The data visualization / dashboard must include the following:
+
+1.    A chart (plot) or charts (plots) that helps to answer the above question.
+2.    A spatial map that helps to answer the above question.
+3.    Make (1) and (2) interactive to allow users to explore the data and uncover insights
+4.    Cross filtering: a filter on the map must update the chart(s) accordingly with the filter, and vice versa.
```
Comment (Gordonei) [inline README.md]:
```suggestion
4.    Cross plot brushing: a filter on the map must update the chart(s) accordingly with the filter, and vice versa.
```

File: README.md [hunk 7]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+5.    Data Storytelling: use the dataset to create a data-driven story, using visualizations to support your narrative.
+6.    Design Principles: Can you explain why you have chosen certain colors (e.g. for legends), fonts, the layout or anything else that will help us understand your thinking in designing the data visualization / dashboard.
+7.    Publish your work using online service like https://pages.github.com/ or usinf any other means you are familiar with.  We want to see the end product and interact with it.
```
Comment (Gordonei) [inline README.md]:
```suggestion
7.    Publish your work using an online service such as https://pages.github.com/ or any other means you are familiar with.  Anyone with an Internet connection and a modern browser such as Google Chrome, Mozilla Firefox or Microsoft Edge, should be able to see the end product and interact with it. Please reference the published link in the `README.md` of your repository.
```

File: README.md
Comment (Gordonei) [inline README.md:1]: I wonder if we should include `2. Initial Data Transformation` for the Data Vis position, too?

If just to test basic scripting and data manipulation?

File: README.md [hunk 8]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+*"In which suburbs should the Water and Sanitation directorate concentrate their infrastructure improvement efforts?".*
+
+The data visualization / dashboard must include the following:
+
+1.    A chart (plot) or charts (plots) that helps to answer the above question.
+2.    A spatial map that helps to answer the above question.
+3.    Make (1) and (2) interactive to allow users to explore the data and uncover insights
+4.    Cross filtering: a filter on the map must update the chart(s) accordingly with the filter, and vice versa.
+5.    Data Storytelling: use the dataset to create a data-driven story, using visualizations to support your narrative.
+6.    Design Principles: Can you explain why you have chosen certain colors (e.g. for legends), fonts, the layout or anything else that will help us understand your thinking in designing the data visualization / dashboard.
```
Comment (Gordonei) [inline README.md]:
```suggestion
6.    Design Principles: In a separate markdown document, please explain why you have chosen certain colours (e.g. for legends), fonts, the layout or anything else that will help us understand your thinking in designing the data visualisation / dashboard to answer the question.
```

**Review by Gordonei — COMMENTED** [review]
First pass

File: README.md (hunk 2 above)
Comment (demainou) [reply README.md]: agree

File: README.md (hunk 3 above)
Comment (demainou) [reply README.md]: agree

File: README.md (hunk 4 above)
Comment (demainou) [reply README.md]: agree

File: README.md
Comment (demainou) [reply README.md:1]: just concerned about the time?

Comment (tockards) [reply README.md:1]: I think its a good idea to add it. It could weigh less than the other tasks when we assess it if your concern is time.

Comment (sulayman-s) [reply README.md:1]: agree with concerns of time, maybe add it as a bonus item at the end of the visualisation challenge? So as not to discourage applicants getting stuck on it at the start.

File: README.md [hunk 9]
```diff
@@ -29,7 +29,7 @@ Over and and above the tasks specified below, there are particular aspects of ea
 * Data Engineer candidates - as the key enablers of our unit's work, we really want to see work done in a sustainable manner: writing for easy comprehension, testing, clean code, modularity all bring us joy.
 * Data Analyst candidates - we consider success for our analysts when they provide the insights that inform actual decisions. Hence, we want evidence of both the ability to surface these insights from data, as well as the rhetorical skill in conveying the implications thereof. Your audience is intelligent, but non-specialist.
 
-### Candidates where programming is required (Data Scientist and Engineers)
+### Candidates where programming is required (Data Scientist; Engineers & Visualisation Engineer)
 Requirements and notes:
 * Our primary programming languages are `python` and `R`. We will accept code that is packaged in `.py`, `.ipynb`, `.R` and `.Rmd` files.
```
Comment (sulayman-s) [inline README.md]: Does this not need to changed to include technologies that visualisation engineer applicant could use to build there visualisation dashboards? Or add a comment for vis engineer to choose an open source technology of their liking. It reads a bit limited now with only `python` and `R`

File: README.md (hunk 4 above)
Comment (ColinAnthony) [reply README.md]: should we ask for an interactive visulaisation/dashboard?

File: README.md
Comment (ColinAnthony) [reply README.md:1]: We do need to test their ability here. `Task 2` is quite a simple join. I can foresee the Vis person needing to some minor transformations, so I support adding it.

File: README.md (hunk 9 above)
Comment (ColinAnthony) [reply README.md]:
```suggestion
* For Data Science and Data Engineering, our primary programming languages are `python`, `R` and `SQL`. We will accept code that is packaged in `.py`, `.ipynb`, `.R` and `.Rmd` files. Scripts in `.sql` may also be included where applicable.
* Data Visualisation engineers should have knowledge of either `python` or `R`, and relevant front-end programming languages (e.g. Javascript, HTML, CSS). We will accept code that is packaged in `.py`, `.R` and appropriate front-end programming language specific files, e.g. `.js`, `.html` etc.
```

File: README.md [hunk 10]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+
+*"In which suburbs should the Water and Sanitation directorate concentrate their infrastructure improvement efforts?".*
+
+The data visualisation / dashboard must include the following:
+
+1.    A chart (plot) or charts (plots) that helps to answer the above question.
+2.    A cartographic map with identifiable landmark features (e.g. major roads, railways, etc.)
+3.    Make (1) and (2) interactive to allow users to explore the data and uncover insights
+4.    Cross plot brushing: a filter on the map must update the chart(s) accordingly with the filter, and vice versa.
+5.    Data Storytelling: in a separate markdown document, use the dataset and your visualisations to outline a data-driven story that answers the above question. In your document, describe how your visualisations support your narrative.
```
Comment (ColinAnthony) [inline README.md]:
```suggestion
5.    Data Storytelling: in a separate markdown document, titled `data-driven-storytelling.md`, provide a brief, step-by-step, point form description of how your visualisations (and information from the dataset) outline a data-driven story that answers the above question.
```

File: README.md [hunk 11]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+6.    Design Principles: In a separate markdown document, please explain why you have chosen certain colours (e.g. for legends), fonts, the layout or anything else that will help us understand your thinking in designing the data visualisation / dashboard to answer the question.
```
Comment (ColinAnthony) [inline README.md]:
```suggestion
6.    Design Principles: In a separate markdown document, titled `visualisation-design-choices.md`, please provide a brief, point form explanation for why you have chosen certain colours (e.g. for legends), fonts, the layout or anything else that will help us understand your thinking in designing the data visualisation / dashboard to answer the question.
```

File: README.md [hunk 12]
```diff
@@ -136,6 +136,22 @@ Please log the time taken to perform the operations described, and within reason
...
+7.    Publish your work using an online service such as https://pages.github.com/ or any other means you are familiar with.  Anyone with an Internet connection and a modern browser such as Google Chrome, Mozilla Firefox or Microsoft Edge, should be able to see the end product and interact with it. Please reference the published link in the `README.md` of your repository.
```
Comment (ColinAnthony) [inline README.md]:
```suggestion
7.    Publish your work using an online service such as https://pages.github.com/ or any other means you are familiar with.  Anyone with an Internet connection and a modern browser such as Google Chrome, Mozilla Firefox or Microsoft Edge, should be able to see the end product and interact with it. Please reference the published link to your visualisation tool in the `README.md` of your repository.
```

File: README.md (hunk 4 above)
Comment (demainou) [reply README.md]: Point 3 ask: *Make (1) and (2) interactive to allow users to explore the data and uncover insights*

File: README.md
Comment (demainou) [reply README.md:1]: Added it as an item to complete and use in subsequent tasks - any objections?

Comment (Gordonei) [reply README.md:1]: I wouldn't require them to use in subsequent tasks - we've actually provided the output of data transformation stuff, so as it not to become a blocker for later tasks  (as @sulayman-s pointed out).

For me, a really good signal is someone getting stuck on something early on, making a good try, then pressing on and completing one of the later ones. Shows stuff about task management and the ability to balance competing tasks (i.e. where they chose to devote their effort).

Comment (demainou) [reply README.md:1]: ok,  is this the correct file for tasks 6 Vis [sr_hex.csv.gz](https://cct-ds-code-challenge-input-data.s3.af-south-1.amazonaws.com/sr_hex.csv.gz) - if confirmed I'll make the change

Comment (demainou) [reply README.md:1]: using [`sr_hex.csv.gz`](https://cct-ds-code-challenge-input-data.s3.af-south-1.amazonaws.com/sr_hex.csv.gz) dataset for tasks 6

File: README.md [hunk 13]
```diff
@@ -90,7 +91,7 @@ Use the [AWS S3 SELECT](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3
 
 Please log the time taken to perform the operations described, and within reason, try to optimise latency and computational resources used. Please also note the comments above about the nature of the code that we expect.
 
-### 2. Initial Data Transformation (if applying for a Data Engineering and/or Science Position)
+### 2. Initial Data Transformation (if applying for a Data Engineering and/or Science Position and Visualisation Engineer)
 Join the equivalent of the contents of the file `city-hex-polygons-8.geojson` to the service request dataset, such that each service request is assigned to a single H3 resolution level 8 hexagon. Use the `sr_hex.csv` file to validate your work.
```
Comment (ColinAnthony) [inline README.md]:
```suggestion
Join the equivalent of the contents of the file `city-hex-polygons-8.geojson` to the service request dataset, such that each service request is assigned to a single H3 resolution level 8 hexagon. Use the `sr_hex.csv.gz` file to validate your work.
```

**Pull request merged**